stistools/add_s_region.py

```diff
@@ -0,0 +1,403 @@
+                        'F25QTZ': '25QTZ',
+                        'F25QTZD1': '25QTZD1',
+                        'F25SRF2': '25SRF2',
+                        'F25SRF2D1': '25SRF2D1',
+                        'F28X50LP': '28X50LP',
+                        'F28X50OII': '28X50O2',
+                        'F28X50OIII': '28X50O3',
+                        'WEDGEA0.6': 'WGA06',
+                        'WEDGEA1.0': 'WGA10',
+                        'WEDGEA1.8': 'WGA18',
+                        'WEDGEA2.0': 'WGA20',
+                        'WEDGEA2.5': 'WGA25',
+                        'WEDGEA2.8': 'WGA28',
+                        'WEDGEB1.0': 'WGB10',
+                        'WEDGEB1.8': 'WGB18',
+                        'WEDGEB2.0': 'WGB20',
+                        'WEDGEB2.5': 'WGB25',
+                        'WEDGEB2.8': 'WGB28'}
+
+h = pysiaf.Siaf('HST')
+
+def get_files_to_process(indir='./'):
+    """Find all _raw.fits and _tag.fits STIS files in the specified directory
+    """
+
+    file_list = []
+    file_list = file_list + glob.glob(os.path.join(indir, 'o*_raw.fits'))
+    file_list = file_list + glob.glob(os.path.join(indir, 'o*_tag.fits'))
+    return file_list
+
+def add_s_region(file, dry_run=False):
+    """Calculate the S_REGION keyword for a single STIS file. If the
+    dry_run parameter is False, set the S_REGION in the SCI extensions with
+    the calculated value.  If keyword isn't present, add it
+    """
+
+    open_mode = 'readonly'
+    if not dry_run: open_mode = 'update'
+    with fits.open(file, mode=open_mode) as f1:
+        log.info('Processing file {}'.format(file))
+        hdr0 = f1[0].header
+        detector = hdr0['DETECTOR']
+        aperture = hdr0['APERTURE']
+        propaper = hdr0['PROPAPER']
+        for ext in f1[1:]:
+            if ext.header['EXTNAME'] == 'SCI':
+                hdr1 = ext.header
+                extver = hdr1['EXTVER']
+                pa_aper = hdr1['PA_APER']
+                pa_aper = pa_aper * DEGREESTORADIANS
+                ra_aper = hdr1['RA_APER']
+                dec_aper = hdr1['DEC_APER']
+                siaf_entry = get_siaf_entry(h, propaper, detector)
+                x, y = siaf_entry.closed_polygon_points('idl')
+                wcslimits = get_wcs_limits(f1)
+                siaflimits = get_siaf_limits(x, y)
+                x, y = smallest_size(wcslimits, siaflimits)
+                # This is to get the parity right
+                x = x * -1.0
+                costheta = math.cos(pa_aper)
+                sintheta = math.sin(pa_aper)
+                dra = x * costheta + y * sintheta
+                dra = dra / math.cos(dec_aper * DEGREESTORADIANS) / 3600.0
+                ddec = (-x * sintheta + y * costheta) / 3600.0
+                ra_corners = ra_aper + dra
+                dec_corners = dec_aper + ddec
+                s_region = 'POLYGON ICRS'
+                for ra, dec in zip(ra_corners, dec_corners):
```

Comment:
Are ra_corners and dec_corners lists?

Reply:
I noticed below that smallest_size returns two lists, so x and y are what ultimately makes *_corners into lists.

Reply:
ra_corners, dec_corners are 1-d numpy arrays.  zip() works on arrays like it does lists (as they are both iterables)